Deploy smart pointers in WebPage.cpp and WebPageMac.mm

rniwa opened the pull request:
#### 69ad827a78b7655646c940bde9fa96d535019145
<pre>
Deploy smart pointers in WebPage.cpp and WebPageMac.mm
<a href="https://bugs.webkit.org/show_bug.cgi?id=260250">https://bugs.webkit.org/show_bug.cgi?id=260250</a>

Reviewed by Wenson Hsieh.

Make more use of smart pointers in these files.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::create):
(WebKit::WebPage::createRemoteSubframe):
(WebKit::WebPage::getFrameInfo):
(WebKit::WebPage::didCommitLoadInAnotherProcess):
(WebKit::WebPage::didFinishLoadInAnotherProcess):
(WebKit::WebPage::frameWasRemovedInAnotherProcess):
(WebKit::WebPage::shouldDispatchSyntheticMouseEventsWhenModifyingSelection const):
(WebKit::WebPage::focusedPluginViewForFrame):
(WebKit::WebPage::pluginViewForFrame):
(WebKit::WebPage::suspendForProcessSwap):
(WebKit::WebPage::loadURLInFrame):
(WebKit::WebPage::loadDataInFrame):
(WebKit::WebPage::loadRequest):
(WebKit::WebPage::navigateToPDFLinkWithSimulatedClick):
(WebKit::WebPage::goToBackForwardItem):
(WebKit::WebPage::tryRestoreScrollPosition):
(WebKit::WebPage::setSize):
(WebKit::WebPage::sendViewportAttributesChanged):
(WebKit::WebPage::scrollMainFrameIfNotAtMaxScrollPosition):
(WebKit::WebPage::drawRect):
(WebKit::WebPage::textZoomFactor const):
(WebKit::WebPage::setTextZoomFactor):
(WebKit::WebPage::pageZoomFactor const):
(WebKit::WebPage::setPageZoomFactor):
(WebKit::WebPage::setPageAndTextZoomFactors):
(WebKit::WebPage::scalePageInViewCoordinates):
(WebKit::WebPage::scaleView):
(WebKit::WebPage::setUseFixedLayout):
(WebKit::WebPage::setFixedLayoutSize):
(WebKit::WebPage::fixedLayoutSize const):
(WebKit::WebPage::updateSizeForCSSDefaultViewportUnits):
(WebKit::WebPage::updateSizeForCSSSmallViewportUnits):
(WebKit::WebPage::updateSizeForCSSLargeViewportUnits):
(WebKit::WebPage::viewportPropertiesDidChange):
(WebKit::WebPage::postInjectedBundleMessage):
(WebKit::WebPage::takeSnapshot):
(WebKit::WebPage::scaledSnapshotWithOptions):
(WebKit::WebPage::snapshotNode):
(WebKit::WebPage::pageDidScroll):
(WebKit::WebPage::pageStoppedScrolling):
(WebKit::WebPage::handleWheelEvent):
(WebKit::WebPage::wheelEvent):
(WebKit::WebPage::dispatchWheelEventWithoutScrolling):
(WebKit::handleKeyEvent):
(WebKit::handleTouchEvent):
(WebKit::WebPage::updatePotentialTapSecurityOrigin):
(WebKit::handleGestureEvent):
(WebKit::WebPage::setBackgroundColor):
(WebKit::WebPage::visibilityDidChange):
(WebKit::WebPage::setNeedsDOMWindowResizeEvent):
(WebKit::WebPage::suspend):
(WebKit::WebPage::runJavaScript):
(WebKit::WebPage::runJavaScriptInFrameInScriptWorld):
(WebKit::WebPage::getContentsAsString):
(WebKit::WebPage::getSelectionAsWebArchiveData):
(WebKit::WebPage::getSelectionOrContentsAsString):
(WebKit::WebPage::getSourceForFrame):
(WebKit::WebPage::getMainResourceDataOfFrame):
(WebKit::resourceDataForFrame):
(WebKit::WebPage::getResourceDataFromFrame):
(WebKit::WebPage::getWebArchiveOfFrame):
(WebKit::WebPage::willCommitLayerTree):
(WebKit::WebPage::sendReportToEndpoints):
(WebKit::WebPage::dragEnded):
(WebKit::WebPage::didStartDrag):
(WebKit::WebPage::dragCancelled):
(WebKit::WebPage::unapplyEditCommand):
(WebKit::WebPage::reapplyEditCommand):
(WebKit::WebPage::unmarkAllMisspellings):
(WebKit::WebPage::unmarkAllBadGrammar):
(WebKit::WebPage::mainFrameHasCustomContentProvider const):
(WebKit::WebPage::updateMainFrameScrollOffsetPinning):
(WebKit::WebPage::mainFrameDidLayout):
(WebKit::WebPage::setMainFrameIsScrollable):
(WebKit::WebPage::SandboxExtensionTracker::shouldReuseCommittedSandboxExtension):
(WebKit::WebPage::beginPrinting):
(WebKit::WebPage::drawToPDF):
(WebKit::WebPage::drawRectToImage):
(WebKit::WebPage::drawPagesToPDFImpl):
(WebKit::pageContainsAnyHorizontalScrollbars):
(WebKit::WebPage::characterIndexForPointAsync):
(WebKit::targetFrameForEditing):
(WebKit::WebPage::cancelComposition):
(WebKit::WebPage::deleteSurrounding):
(WebKit::WebPage::setAlwaysShowsHorizontalScroller):
(WebKit::WebPage::setAlwaysShowsVerticalScroller):
(WebKit::WebPage::setMinimumSizeForAutoLayout):
(WebKit::WebPage::setSizeToContentAutoSizeMaximumSize):
(WebKit::WebPage::setAutoSizingShouldExpandToViewHeight):
(WebKit::WebPage::setViewportSizeForCSSViewportUnits):
(WebKit::WebPage::didCommitLoad):
(WebKit::WebPage::updateAttachmentAttributes):
(WebKit::WebPage::updateAttachmentThumbnail):
(WebKit::WebPage::updateAttachmentIcon):
(WebKit::WebPage::requestAttachmentIcon):
(WebKit::WebPage::attachmentElementWithIdentifier const):
(WebKit::WebPage::getApplicationManifest):
(WebKit::WebPage::getTextFragmentMatch):
(WebKit::WebPage::contextForElement const):
(WebKit::WebPage::startTextManipulationForFrame):
(WebKit::WebPage::configureLoggingChannel):
(WebKit::WebPage::updateWithTextRecognitionResult):
(WebKit::WebPage::createMediaSessionCoordinator):
(WebKit::WebPage::lastNavigationWasAppInitiated):
(WebKit::WebPage::scrollToRect):
(WebKit::WebPage::generateTestReport):
(WebKit::WebPage::useRedirectionForCurrentNavigation):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::handleAcceptedCandidate):
(WebKit::WebPage::executeKeypressCommandsInternal):
(WebKit::WebPage::handleEditingKeyboardEvent):
(WebKit::WebPage::attributedSubstringForCharacterRangeAsync):
(WebKit::WebPage::performNonEditingBehaviorForSelector):
(WebKit::WebPage::getStringSelectionForPasteboard):
(WebKit::WebPage::getDataSelectionForPasteboard):
(WebKit::WebPage::shouldDelayWindowOrderingEvent):
(WebKit::WebPage::requestAcceptsFirstMouse):
(WebKit::WebPage::setTopOverhangImage):
(WebKit::WebPage::setBottomOverhangImage):
(WebKit::WebPage::computePagesForPrintingPDFDocument):
(WebKit::WebPage::performImmediateActionHitTestAtLocation):
(WebKit::WebPage::lookupTextAtLocation):
(WebKit::WebPage::immediateActionDidUpdate):
(WebKit::WebPage::immediateActionDidCancel):
(WebKit::WebPage::immediateActionDidComplete):
(WebKit::WebPage::dataDetectorsDidHideUI):
(WebKit::WebPage::didEndMagnificationGesture):

Canonical link: <a href="https://commits.webkit.org/266958@main">https://commits.webkit.org/266958@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1dd783b741939c08fe0ec4990b367d019959750d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/15231 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/15536 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15896 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16987 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/14308 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/18051 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/15634 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/16919 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/15414 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15852 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12952 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/17720 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/13132 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/20702 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/14212 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13908 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/17165 "2 api tests failed or timed out") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/14476 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/12269 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13751 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/18095 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/1846 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/14313 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
```diff
@@ -653,9 +653,9 @@ static String commandNameForSelectorName(const String& selectorName)
-    RetainPtr<PDFDocument> pdfDocument = coreFrame ? pdfDocumentForPrintingFrame(coreFrame) : 0;
+    RefPtr frame = WebProcess::singleton().webFrame(frameID);
+    RefPtr coreFrame = frame ? frame->coreLocalFrame() : nullptr;
+    RetainPtr<PDFDocument> pdfDocument = coreFrame ? pdfDocumentForPrintingFrame(coreFrame.get()) : 0;
```
Nit - maybe we can just replace `0` with `nil` while we're here?

rniwa committed:
Deploy smart pointers in WebPage.cpp and WebPageMac.mm
https://bugs.webkit.org/show_bug.cgi?id=260250

Reviewed by Wenson Hsieh.

Make more use of smart pointers in these files.

* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::WebPage::create):
(WebKit::WebPage::createRemoteSubframe):
(WebKit::WebPage::getFrameInfo):
(WebKit::WebPage::didCommitLoadInAnotherProcess):
(WebKit::WebPage::didFinishLoadInAnotherProcess):
(WebKit::WebPage::frameWasRemovedInAnotherProcess):
(WebKit::WebPage::shouldDispatchSyntheticMouseEventsWhenModifyingSelection const):
(WebKit::WebPage::focusedPluginViewForFrame):
(WebKit::WebPage::pluginViewForFrame):
(WebKit::WebPage::suspendForProcessSwap):
(WebKit::WebPage::loadURLInFrame):
(WebKit::WebPage::loadDataInFrame):
(WebKit::WebPage::loadRequest):
(WebKit::WebPage::navigateToPDFLinkWithSimulatedClick):
(WebKit::WebPage::goToBackForwardItem):
(WebKit::WebPage::tryRestoreScrollPosition):
(WebKit::WebPage::setSize):
(WebKit::WebPage::sendViewportAttributesChanged):
(WebKit::WebPage::scrollMainFrameIfNotAtMaxScrollPosition):
(WebKit::WebPage::drawRect):
(WebKit::WebPage::textZoomFactor const):
(WebKit::WebPage::setTextZoomFactor):
(WebKit::WebPage::pageZoomFactor const):
(WebKit::WebPage::setPageZoomFactor):
(WebKit::WebPage::setPageAndTextZoomFactors):
(WebKit::WebPage::scalePageInViewCoordinates):
(WebKit::WebPage::scaleView):
(WebKit::WebPage::setUseFixedLayout):
(WebKit::WebPage::setFixedLayoutSize):
(WebKit::WebPage::fixedLayoutSize const):
(WebKit::WebPage::updateSizeForCSSDefaultViewportUnits):
(WebKit::WebPage::updateSizeForCSSSmallViewportUnits):
(WebKit::WebPage::updateSizeForCSSLargeViewportUnits):
(WebKit::WebPage::viewportPropertiesDidChange):
(WebKit::WebPage::postInjectedBundleMessage):
(WebKit::WebPage::takeSnapshot):
(WebKit::WebPage::scaledSnapshotWithOptions):
(WebKit::WebPage::snapshotNode):
(WebKit::WebPage::pageDidScroll):
(WebKit::WebPage::pageStoppedScrolling):
(WebKit::WebPage::handleWheelEvent):
(WebKit::WebPage::wheelEvent):
(WebKit::WebPage::dispatchWheelEventWithoutScrolling):
(WebKit::handleKeyEvent):
(WebKit::handleTouchEvent):
(WebKit::WebPage::updatePotentialTapSecurityOrigin):
(WebKit::handleGestureEvent):
(WebKit::WebPage::setBackgroundColor):
(WebKit::WebPage::visibilityDidChange):
(WebKit::WebPage::setNeedsDOMWindowResizeEvent):
(WebKit::WebPage::suspend):
(WebKit::WebPage::runJavaScript):
(WebKit::WebPage::runJavaScriptInFrameInScriptWorld):
(WebKit::WebPage::getContentsAsString):
(WebKit::WebPage::getSelectionAsWebArchiveData):
(WebKit::WebPage::getSelectionOrContentsAsString):
(WebKit::WebPage::getSourceForFrame):
(WebKit::WebPage::getMainResourceDataOfFrame):
(WebKit::resourceDataForFrame):
(WebKit::WebPage::getResourceDataFromFrame):
(WebKit::WebPage::getWebArchiveOfFrame):
(WebKit::WebPage::willCommitLayerTree):
(WebKit::WebPage::sendReportToEndpoints):
(WebKit::WebPage::dragEnded):
(WebKit::WebPage::didStartDrag):
(WebKit::WebPage::dragCancelled):
(WebKit::WebPage::unapplyEditCommand):
(WebKit::WebPage::reapplyEditCommand):
(WebKit::WebPage::unmarkAllMisspellings):
(WebKit::WebPage::unmarkAllBadGrammar):
(WebKit::WebPage::mainFrameHasCustomContentProvider const):
(WebKit::WebPage::updateMainFrameScrollOffsetPinning):
(WebKit::WebPage::mainFrameDidLayout):
(WebKit::WebPage::setMainFrameIsScrollable):
(WebKit::WebPage::SandboxExtensionTracker::shouldReuseCommittedSandboxExtension):
(WebKit::WebPage::beginPrinting):
(WebKit::WebPage::drawToPDF):
(WebKit::WebPage::drawRectToImage):
(WebKit::WebPage::drawPagesToPDFImpl):
(WebKit::pageContainsAnyHorizontalScrollbars):
(WebKit::WebPage::characterIndexForPointAsync):
(WebKit::targetFrameForEditing):
(WebKit::WebPage::cancelComposition):
(WebKit::WebPage::deleteSurrounding):
(WebKit::WebPage::setAlwaysShowsHorizontalScroller):
(WebKit::WebPage::setAlwaysShowsVerticalScroller):
(WebKit::WebPage::setMinimumSizeForAutoLayout):
(WebKit::WebPage::setSizeToContentAutoSizeMaximumSize):
(WebKit::WebPage::setAutoSizingShouldExpandToViewHeight):
(WebKit::WebPage::setViewportSizeForCSSViewportUnits):
(WebKit::WebPage::didCommitLoad):
(WebKit::WebPage::updateAttachmentAttributes):
(WebKit::WebPage::updateAttachmentThumbnail):
(WebKit::WebPage::updateAttachmentIcon):
(WebKit::WebPage::requestAttachmentIcon):
(WebKit::WebPage::attachmentElementWithIdentifier const):
(WebKit::WebPage::getApplicationManifest):
(WebKit::WebPage::getTextFragmentMatch):
(WebKit::WebPage::contextForElement const):
(WebKit::WebPage::startTextManipulationForFrame):
(WebKit::WebPage::configureLoggingChannel):
(WebKit::WebPage::updateWithTextRecognitionResult):
(WebKit::WebPage::createMediaSessionCoordinator):
(WebKit::WebPage::lastNavigationWasAppInitiated):
(WebKit::WebPage::scrollToRect):
(WebKit::WebPage::generateTestReport):
(WebKit::WebPage::useRedirectionForCurrentNavigation):
* Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
(WebKit::WebPage::handleAcceptedCandidate):
(WebKit::WebPage::executeKeypressCommandsInternal):
(WebKit::WebPage::handleEditingKeyboardEvent):
(WebKit::WebPage::attributedSubstringForCharacterRangeAsync):
(WebKit::WebPage::performNonEditingBehaviorForSelector):
(WebKit::WebPage::getStringSelectionForPasteboard):
(WebKit::WebPage::getDataSelectionForPasteboard):
(WebKit::WebPage::shouldDelayWindowOrderingEvent):
(WebKit::WebPage::requestAcceptsFirstMouse):
(WebKit::WebPage::setTopOverhangImage):
(WebKit::WebPage::setBottomOverhangImage):
(WebKit::WebPage::computePagesForPrintingPDFDocument):
(WebKit::WebPage::performImmediateActionHitTestAtLocation):
(WebKit::WebPage::lookupTextAtLocation):
(WebKit::WebPage::immediateActionDidUpdate):
(WebKit::WebPage::immediateActionDidCancel):
(WebKit::WebPage::immediateActionDidComplete):
(WebKit::WebPage::dataDetectorsDidHideUI):
(WebKit::WebPage::didEndMagnificationGesture):

Canonical link: https://commits.webkit.org/266958@main

szewai commented on Source/WebKit/WebProcess/WebPage/WebPage.cpp:
```diff
@@ -8065,7 +8066,7 @@ void WebPage::insertAttachment(const String& identifier, std::optional<uint64_t>
 void WebPage::updateAttachmentAttributes(const String& identifier, std::optional<uint64_t>&& fileSize, const String& contentType, const String& fileName, const IPC::SharedBufferReference& enclosingImageData, CompletionHandler<void()>&& callback)
 {
-    if (auto attachment = attachmentElementWithIdentifier(identifier)) {
+    if (RefPtr attachment = attachmentElementWithIdentifier(identifier)) {
```
`attachmentElementWithIdentifier` already returns `RefPtr`

rniwa replied:
Even then, it's good to document the fact it's a RefPtr by explicitly stating it.

szewai replied:
But for the result of `targetFrameForEditing` you used `auto` in this patch...

rniwa replied:
Oh, I forgot to fix those :( We really need clang static analyzer to enforce the rules rather than me following the rules manually.

szewai commented on Source/WebKit/WebProcess/WebPage/mac/WebPageMac.mm:
```diff
@@ -847,28 +847,28 @@ static void drawPDFPage(PDFDocument *pdfDocument, CFIndex pageIndex, CGContextRe
 
-    auto& mainFrame = *localMainFrame;
-    if (!mainFrame.view() || !mainFrame.view()->renderView()) {
+    Ref mainFrame = *localMainFrame;
```
Why do we need to create both RefPtr and Ref here?

rniwa replied:
Oh, indeed, that was unnecessary.
